frontend/components/common/FlexScrollGradient.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+   Flex,
+   FlexProps,
+   SystemStyleObject,
+   useBreakpointValue,
+   useMergeRefs,
+} from '@chakra-ui/react';
+import { useRef, useLayoutEffect, forwardRef, useState } from 'react';
+
+type InnerFlexStyling = {
+   _after: SystemStyleObject;
+   _before: SystemStyleObject;
+   overflowX?: FlexProps['overflowX'];
+   overflowY?: FlexProps['overflowY'];
+};
+
+const gradientWidths = { base: 40, sm: 50, md: 70, lg: 100 };
+
+const sharedStyle: SystemStyleObject = {
+   content: '""',
+   position: 'absolute',
+   height: '100%',
+   pointerEvents: 'none',
+   zIndex: 9999,
+};
+
+function easeInOutQuad(x: number): number {
+   return x < 0.5 ? 2 * x * x : 1 - Math.pow(-2 * x + 2, 2) / 2;
+}
+
+enum FlexScrollDirection {
+   ROW,
+   COLUMN,
+}
+
+function getScrollDirection(el: HTMLElement): FlexScrollDirection {
+   const { flexDirection } = window.getComputedStyle(el);
+   if (flexDirection === 'column') {
+      return FlexScrollDirection.COLUMN;
+   } else {
+      return FlexScrollDirection.ROW;
+   }
+}
+
+function getGradientsBaseStyle(
```

Comment:
It's hard to tell exactly what this component does...  but it feels very complex for what seemingly amounts to making a gradient slightly bigger or smaller depending on the browser width. I don't doubt that it looks cool but it strikes me as a lot of complexity (and thus the maintenance) for what it buys us.

Reply:
> This is copied from code in the monorepo.

Ok. Though that doesn't refute the point I was trying to make. Potentially makes it worse as we have at least two copies of this complexity.

> Although, I hear you, we have this implemented in a few different ways in the mono repo and all versions behave differently

[:trollface:](https://xkcd.com/927/)

> I can port the full test suite for this component across as well. If that helps?

I'd rather not. It doesn't help my concerns of complexity.

Reply:
When examining how this looks, I think I prefer the look of a fixed-height gradient (or none at all) over one that has a dynamic height.

Reply:
I think some more general discussion about the desired fade behavior might be in order.

The other feature taken into account here is that the fade slowly shifts its opacity as you scroll until you pass its end.

Here is what I am talking about:

[Screencast from 2023-08-22 16-59-54.webm](https://github.com/iFixit/react-commerce/assets/857362/836c87d4-6ab0-4cae-89b0-452731f0d71e)

Note: I made the fade red to help visibility.

Here's what it looks like normally

[Screencast from 2023-08-22 17-01-35.webm](https://github.com/iFixit/react-commerce/assets/857362/1a97e5f2-34ca-46b7-9a87-0a97b27cb2b3)

Reply:
@ianrohde Showed me this approach which does work, but only in chrome and it puts the box shadow behind text, not in front of it: https://daverupert.com/2023/08/animation-timeline-scroll-shadows/